src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        int[] employeesSalaries = new int[names.length];
+
+        for (String record : data) {
+            String[] partsOfData = record.split(" ");
+            String dataDate = partsOfData[0];
+            LocalDate currentDate = LocalDate.parse(dataDate, formatter);
+            String dataName = partsOfData[1];
+            int dataHoursWorked = Integer.parseInt(partsOfData[2]);
+            int dataRatePerHour = Integer.parseInt(partsOfData[3]);
+
+            if (currentDate.isEqual(startDate) || currentDate.isAfter(startDate)) {
```

Comment:
You can simplify the if-statement by using `!currentDate.isBefore(startDate)` instead of `currentDate.isEqual(startDate) || currentDate.isAfter(startDate)`.